mirrord/cli/src/main.rs

```diff
@@ -48,36 +48,25 @@ use mac::temp_dir;
 
 fn extract_library(dest_dir: Option<String>) -> Result<PathBuf> {
     let progress = TaskProgress::new("initializing mirrord layer...");
-    let library_file = env!("MIRRORD_LAYER_FILE");
-    let library_path = Path::new(library_file);
-
-    let extension = library_path
-        .components()
-        .last()
+    let extension = Path::new(env!("MIRRORD_LAYER_FILE"))
+        .extension()
         .unwrap()
-        .as_os_str()
         .to_str()
-        .unwrap()
-        .split('.')
-        .collect::<Vec<&str>>()[1];
-
-    let file_name = format!(
-        "{}-libmirrord_layer.{extension}",
-        Alphanumeric
-            .sample_string(&mut rand::thread_rng(), 10)
-            .to_lowercase()
-    );
+        .unwrap();
 
+    let file_name = format!("{}-libmirrord_layer.{extension}", const_random!(u64));
     let file_path = match dest_dir {
         Some(dest_dir) => std::path::Path::new(&dest_dir).join(file_name),
         None => temp_dir().as_path().join(file_name),
     };
-    let mut file = File::create(&file_path)
-        .with_context(|| format!("Path \"{}\" creation failed", file_path.display()))?;
-    let bytes = include_bytes!(env!("MIRRORD_LAYER_FILE"));
-    file.write_all(bytes).unwrap();
+    if !file_path.exists() {
```

Comment:
Should we maybe, only when creating a new layer file because it does not yet exist in this version, also delete the last one? That way we avoid a filling up the user's drive at a later point, but they still get a quick run without copying files most of the time - when they are rerunning the same mirrord binary.

Reply:
I think it's hard to solve and also not very beneficial - how many updates does a user do?

Reply:
Why's it hard to solve? Don't we just remove all *-libmirrord_layer.{extension} from the temp_dir?

Reply:
Nevermind merging